### PR TITLE
[geoclue-provider-mlsdb] Prefer location from online MLS if possible.…

### PR DESCRIPTION
**Pull request by blammit**
… Fixes JB#34847

Add MlsdbOnlineLocator, which fetches the current location from the
online MLS using a combination of cellular and WiFi data.

### CODE REVIEW

File: plugin/mlsdbonlinelocator.cpp [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+        QVariantMap cellTowerMap;
+        // supported radio types: gsm, wcdma or lte
+        switch (cell.uniqueCellId.cellType()) {
+        case MLSDB_CELL_TYPE_LTE:
+            cellTowerMap["radioType"] = "lte";
+            break;
+        case MLSDB_CELL_TYPE_GSM:
+            cellTowerMap["radioType"] = "gsm";
+            break;
+        default:
```
Comment (chriadam) [inline plugin/mlsdbonlinelocator.cpp]: In the MLS database, WCDMA are reported as UMTS I believe.  So you can do case MLSDB_CELL_TYPE_UMTS: cellTowerMap["radioType"] = "wcdma"; break; I believe.

File: plugin/mlsdbonlinelocator.cpp [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+    }
+    map["wifiAccessPoints"] = wifiInfoList;
+    return map;
+}
+
+QVariantMap MlsdbOnlineLocator::fallbackFields()
+{
+    QVariantMap fallbacks;
+    fallbacks["lacf"] = true;
+    fallbacks["ipf"] = true;
```
Comment (chriadam) [inline plugin/mlsdbonlinelocator.cpp]: Can a comment be included to describe what these are?

File: plugin/mlsdbprovider.cpp [hunk 3]
```diff
@@ -286,6 +287,52 @@ void MlsdbProvider::timerEvent(QTimerEvent *event)
 
 void MlsdbProvider::calculatePositionAndEmitLocation()
 {
+    tryFetchOnlinePosition();
```
Comment (chriadam) [inline plugin/mlsdbprovider.cpp]: This feature needs to be selectable by the user.  We need to add a new location setting key (different from the current cell_id_positioning one) which allows the user to enable or disable the online MLS lookup.  Both for privacy reasons and battery/data use reasons.

**Comment by chriadam** [comment]
Looks great, just a couple of comments.  Note that we're going to completely change the format of the /etc/location/location.conf file, so that it will be separated into provider-specific sections, but we should discuss that offline.

**Comment by blammit** [comment]
Thanks, updated as per comments. Just reads an additional field from /etc/location/location.conf for now, if it's not being changed yet.

**Comment by chriadam** [comment]
@sage @slava @pvuorela should review this too

**Comment by chriadam** [comment]
Also worth linking this PR into MER#1559 I think.

**Comment by pvuorela** [comment]
Wonder how many apps request small interval for getting updates. Like any chance for timer triggering every second and then resulting in network request? If that's a problem, could perhaps force some limitation how often location is queried or then cache latest query and omit duplicate ones. Though latter can have problems with volatile ap list and cellular signal strengths.

Generally LGTM.

**Comment by Slava** [comment]
I think you tagged the wrong person.

**Comment by pvuorela** [comment]
ping @monich :D

**Comment by pvuorela** [comment]
and @saukko perhaps better too.

**Comment by chriadam** [comment]
Obsoleted by https://github.com/mer-hybris/geoclue-providers-mlsdb/pull/8

**Pull request closed without merging**